### PR TITLE
[Draft] Respect max-age header of public keys response

### PR DESCRIPTION
**Pull request by lucasavila00**
I'm not sure if this is the best way to parse the tokens, I just copied the code from https://github.com/firebase/firebase-admin-go/blob/master/auth/token_verifier.go

And I just realized I based this branch on my other PR.

Anyway, I wouldn't mind this PR being closed in favor of a better one that does the same thing, and I'd be thrilled to know what the proper code would look like :smile:

### CODE REVIEW

**Comment by Nickforall** [comment]
Woah this is awesome

File: lib/source/google_key_source.ex [hunk 1]
```diff
@@ -26,4 +26,29 @@ defmodule ExFirebaseAuth.KeySource.Google do
...
+
+  defp get_max_age(headers) do
+    headers |> get_cache_control_header() |> get_get_max_age_from_cache_control()
+  end
+
+  defp get_cache_control_header(headers) do
+    Enum.find(headers, fn {k, _v} -> k == "cache-control" end)
+  end
+
+  defp get_get_max_age_from_cache_control(nil) do
```
Comment (Nickforall) [inline lib/source/google_key_source.ex:38]: It looks like you made a small typo here

```suggestion
  defp get_max_age_from_cache_control(nil) do
```

File: lib/source/google_key_source.ex [hunk 2]
```diff
@@ -26,4 +26,29 @@ defmodule ExFirebaseAuth.KeySource.Google do
...
+    300
+  end
+
+  defp get_get_max_age_from_cache_control({"cache-control", value}) do
+    values = String.split(value, ",")
+
+    Enum.find_value(values, fn x ->
```
Comment (Nickforall) [inline lib/source/google_key_source.ex:45]: `Enum.find_value/2` returns a default, which is currently nil but I think we should make that 300 as well (https://hexdocs.pm/elixir/Enum.html#find_value/3)

https://hexdocs.pm/elixir/Enum.html#find_value/3

**Review by Nickforall — CHANGES_REQUESTED** [review]
Looks good, some small comments but nice work!

**Comment by lucasavila00** [comment]
I want to re-do this PR from scratch. I'll try do it tomorrow.

**Comment by Nickforall** [comment]
Sure thing! No worries :)

Feel free to close this PR and open another one or continue on this PR :)

**Comment by Nickforall** [comment]
Closing this as stale

**Pull request closed without merging**